tests/tt_eager/python_api_testing/unit_testing/misc/test_reshard.py

```diff
@@ -343,3 +343,125 @@ def test_reshard_with_program_cache(
     assert passing, output
 
     assert device.num_program_cache_entries() == 3
+
```

Comment:
I think we want to eventually move stuff out of this directory. 
would it be possible to add the tests in `tests/ttnn/unit_tests/operations/test_core.py` . There are a bunch of reshard tests there (its using `to_memory_config`)

Reply:
I think there should be separate tests for calling `reshard` directly and `to_memory_config` since calling `to_memory_config` doesn't guarantee that it will invoke `reshard` or could change the arguments.